src/api/users/notifications.js

```diff
@@ -0,0 +1,166 @@
+                console.error('Parâmetros inválidos para enviar notificação.');
+                return;
+            }
+            const userDataRef = await getDoc(doc(db, 'users', followerId));
+            existes(userDataRef)
+            const notificationData = {
+                userId: followedId,
+                followerId: followerId,
+                data: {
+                    title: `${userDataRef.data().name}`,
+                    action: 'Começou a seguir você!',
+                    isRead: false,
+                },
+                time,
+            };
+            try {
+                await addDoc(notificationsCollection, notificationData);
+            } catch (error) {
+                console.error('Erro ao adicionar notificação:', error);
+            }
+        },
+        postUnfollow: async (followedId, followerId) => {
+            if (!followerId || !followedId) {
+                console.error('Parâmetros inválidos para enviar notificação.');
+                return;
+            }
+            const userDataRef = await getDoc(doc(db, 'users', followerId));
+            existes(userDataRef)
+            try {
+                const docRef = await addDoc(notificationsCollection, {
+                    userId: followedId,
+                    unfollowerId: userDataRef.id,
+                    data: {
+                        title: `${userDataRef.data().name}`,
+                        action: 'Deixou de seguir você.',
+                        isRead: false,
+                    },
+                    time,
+                });
+                console.log('Notificação de unfollow adicionada com sucesso:', docRef.id);
+            } catch (error) {
+                console.error('Erro ao adicionar notificação de unfollow:', error);
+            }
+        },
+        hasAlreadyBeenSeen: async (userId, notificationId) => {
+            if (!userId || !notificationId) return;
+            try {
+                const notificationsRef = collection(db, 'notifications');
+                const querySnapshot = await getDocs(query(notificationsRef, where('userId', '==', userId)));
+                querySnapshot.forEach(async (doc) => {
+                    const data = doc.data();
+                    const docId = doc.ref.id;
+                    if (docId === notificationId) {
+                        await updateDoc(doc.ref, {
+                            data: {
+                                ...data.data,
+                                isRead: true,
+                                alreadyOpenedHours: new Date().toLocaleTimeString(),
+                                alreadyOpenedDate: new Date().toLocaleDateString(),
+                            },
+                        });
+                    }
+                });
+            } catch (error) {
+                console.error('Erro ao verificar se as notificações foram lidas:', error);
+                return false;
+            }
+        },
```

Comment:
essa função é pra marcar uma única notificação como lida?

Reply:
isso, essa função marca uma not... como lida, também adiciona alguns campos no firebase como: alreadyOpenedHours e alreadyOpenedHours,  elém disso essa função retorna todas as notificações que o userId é igual ao id do usuário fornecido e no final filtra essas notificações pelo id da notificação selecionada.